fbw-a380x/src/wasm/systems/a380_systems/src/hydraulic/autobrakes.rs

```diff
@@ -541,5 +604,777 @@ impl SimulationElement for A380AutobrakeController {
+
+            deceleration_request: Acceleration::default(),
+            end_of_decel_acceleration: Acceleration::default(),
+            desired_deceleration: Acceleration::new::<meter_per_second_squared>(
+                Self::MAX_DECEL_DRY_MS2,
+            ),
+            actual_deceleration: Acceleration::default(),
+
+            final_distance_remaining: Length::default(),
+
+            distance_remaining_at_decel_activation: Length::default(),
+
+            dry_prediction: Length::default(),
+            wet_prediction: Length::default(),
+
+            distance_to_rwy_end: Length::default(),
+        }
+    }
+
+    fn enable(&mut self) {
+        if self.state == BTVState::Disabled && self.arming_authorized() {
+            self.state = BTVState::Armed;
+        }
+    }
+
+    fn disarm(&mut self) {
+        self.state = BTVState::Disabled;
+        self.deceleration_request = Acceleration::new::<meter_per_second_squared>(5.);
+        self.end_of_decel_acceleration = Acceleration::new::<meter_per_second_squared>(5.);
+        self.final_distance_remaining = Length::default();
+        self.distance_remaining_at_decel_activation = Length::default();
+        self.desired_deceleration =
+            Acceleration::new::<meter_per_second_squared>(Self::MAX_DECEL_DRY_MS2);
+    }
+
+    fn decel(&self) -> Acceleration {
+        match self.state {
+            BTVState::Decel | BTVState::_OutOfDecelRange => self.deceleration_request,
+            BTVState::EndOfBraking => self.end_of_decel_acceleration,
+            BTVState::RotOptimization => self.accel_during_rot_opti(),
+            _ => Acceleration::new::<meter_per_second_squared>(5.),
+        }
+    }
+
+    fn update(
+        &mut self,
+        context: &UpdateContext,
+        spoilers_active: bool,
+        braking_distance: &BrakingDistanceCalculator,
+        ground_speed: Velocity,
+        rop: &AutobrakeRunwayOverrunProtection,
+    ) {
+        self.distance_to_rwy_end = rop.distance_to_runway_end();
+
+        self.wet_prediction = braking_distance.wet_landing();
+        self.dry_prediction = braking_distance.dry_landing();
+
+        self.spoilers_active = spoilers_active;
+        self.actual_deceleration = context.long_accel();
+
+        self.integrate_distance(context, ground_speed);
+
+        self.compute_decel(ground_speed);
+
+        self.state = self.update_state(ground_speed);
+    }
+
+    fn braking_distance_remaining(&self) -> Length {
+        let distance_remaining_raw = if self.is_oans_fallback_mode() {
+            self.in_flight_btv_stopping_distance.value() - self.rolling_distance
+        } else {
+            self.oans_distance_to_exit.value()
+        };
+
+        let distance_from_btv_exit = Length::new::<meter>(Self::DISTANCE_OFFSET_TO_RELEASE_BTV_M);
+
+        // Max distance clamped to end of rwy minus a margin
+        (distance_remaining_raw - distance_from_btv_exit)
+            .max(Length::default())
+            .min(
```

Comment:
Use `clamp`

Reply:
I see. You could add something like this into fbw_common/shared. If you want I can commit it to your branch.
```rust
trait Clamp {
    /// Restrict a value to a certain interval unless it is NaN.
    ///
    /// Returns `max` if `self` is greater than `max`, and `min` if `self` is
    /// less than `min`. Otherwise this returns `self`.
    ///
    /// Note that this function returns NaN if the initial value was NaN as
    /// well.
    ///
    /// # Panics
    ///
    /// Panics if `min > max`, `min` is NaN, or `max` is NaN.
    fn clamp(self, min: Self, max: Self) -> Self;
}

// TODO: remove when uom implements clamp on floating point quantities
impl<D: uom::si::Dimension + ?Sized, U: uom::si::Units<f64> + ?Sized> Clamp
    for Quantity<D, U, f64>
{
    fn clamp(mut self, min: Self, max: Self) -> Self {
        assert!(
            min <= max,
            "min > max, or either was NaN. min = {min:?}, max = {max:?}"
        );
        if self < min {
            self = min;
        }
        if self > max {
            self = max;
        }
        self
    }
}
```